Segementation volume and centroid calculation

plt-joey opened the pull request:
Addresses #1894 and #1895

~~Currently missing unit tests which will be added soon.~~ Unit tests added.

ivmartel commented:
Looking good, thanks!

I was wondering how to make the feature easier to use, maybe as a method on the image class to update the volumes and centroids. This method would be manually called from the brush tool at mouse up and would fire an event once volumes have been re-calculated.

Could you add a 'go to segment' button to the demo viewer? Something similar to the 'go to annotation' of annotations. The button would go to the first centroid. This would allow to check if values are correct.

plt-joey commented:
> Looking good, thanks!
> 
> I was wondering how to make the feature easier to use, maybe as a method on the image class to update the volumes and centroids. This method would be manually called from the brush tool at mouse up and would fire an event once volumes have been re-calculated.
> 
> Could you add a 'go to segment' button to the demo viewer? Something similar to the 'go to annotation' of annotations. The button would go to the first centroid. This would allow to check if values are correct.

Simplified the feature so now you simply have to watch an event on the brush. You can also manually trigger volume calculations with `Brush.recalculateVolumes`.

Also added a "Goto first volume" button (and fixed a minor bug in the centroid calculation)

Let me know what you think.

ivmartel commented:
I would prefer the volume calculation to be available from the image class. That way you do not have to depend on the brush tool to calculate it and can provide volume and centroid after loading a mask whatever the activated tool.

plt-joey commented:
> I would prefer the volume calculation to be available from the image class. That way you do not have to depend on the brush tool to calculate it and can provide volume and centroid after loading a mask whatever the activated tool.

Moved everything into `Image`. Had to be made a bit less memory efficient (requires a separate `Volumes` object for every image that needs volumes) but it is much cleaner.

ivmartel commented:
Regarding the `onVolumeCalculation` method, why don't you just send an event (image has a `#fireEvent`)?
I checked the goto volume button, would it be complicated to have a goto segment instead? Could the volumes be index by segment number?

plt-joey commented:
> Regarding the `onVolumeCalculation` method, why don't you just send an event (image has a `#fireEvent`)? I checked the goto volume button, would it be complicated to have a goto segment instead? Could the volumes be indexed by segment number?

Didn't notice `#fireEvent` (I would have expected it to extend `EventTarget` for that), I have removed the callback and used that instead. Also added the segment number to the volumes and changed the goto button to be per-segment.

plt-joey commented:
> Last 2 comments, I swear! Regarding the volumes, I think it could be nice to order them by volume for each segment, that way the first one of the segment is the biggest and the goto goes to that one. Regarding tests, the assert methods do accept a string parameter to help identify the test, could you add them please?

sorted and named
